forum/users/permissions.py

```diff
@@ -0,0 +1,29 @@
+from rest_framework.permissions import BasePermission
+from rest_framework.request import Request
+from .models import Role, CustomUser
+
+class HasRole(BasePermission):
+    """
+    Permission class to check if a user has the required role,
+    based on both the token and the database state.
+    """
+
+    def __init__(self, role: Role):
+        self.role = role
+
+    def has_permission(self, request: Request, view) -> bool:
+        if not request.user.is_authenticated:
+            return False
+
+        try:
+            token_role = request.auth.get('role')
```

Comment:
It’s better to explicitly verify the presence of `request.auth` and its structure before accessing it

Reply:
Done